File: app/src/test/java/com/khmelenko/lab/travisclient/TestTimeConverter.java
package com.khmelenko.lab.travisclient;

import com.khmelenko.lab.travisclient.converter.TimeConverter;

import org.junit.Assert;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Testing TimeConverter class
 *
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
public class TestTimeConverter {

    @Test
    public void testDurationToString() {

        long zeroDuration = 0;
        long onlySeconds = 45;
        long onlyMinutes = 5 * 60;
        long onlyHours = 3 * 60 * 60;
        long secondsAndMinutes = onlyMinutes + onlySeconds;
        long hoursMinutesAndSeconds = onlyHours + secondsAndMinutes;
        long hoursAndSeconds = onlyHours + onlySeconds;
        long hoursAndMinutes = onlyHours + onlyMinutes;

        String zeroDurationString = "0 sec";
        String onlySecondsString = "45 sec";
        String onlyMinutesString = "5 min";
        String onlyHoursString = "3 hours";
        String secondsAndMinutesString = String.format("%s %s", onlyMinutesString, onlySecondsString);
        String hoursMinutesAndSecondsString = String.format("%s %s %s", onlyHoursString, onlyMinutesString, onlySecondsString);
        String hoursAndSecondsString = String.format("%s %s", onlyHoursString, onlySecondsString);
        String hoursAndMinutesString = String.format("%s %s", onlyHoursString, onlyMinutesString);

        String actual = TimeConverter.durationToString(onlySeconds);
        assertEquals(onlySecondsString, actual);

        actual = TimeConverter.durationToString(onlyMinutes);
        assertEquals(onlyMinutesString, actual);

        actual = TimeConverter.durationToString(onlyHours);
        assertEquals(onlyHoursString, actual);

        actual = TimeConverter.durationToString(secondsAndMinutes);
        assertEquals(secondsAndMinutesString, actual);

        actual = TimeConverter.durationToString(hoursMinutesAndSeconds);
        assertEquals(hoursMinutesAndSecondsString, actual);

        actual = TimeConverter.durationToString(hoursAndSeconds);
        assertEquals(hoursAndSecondsString, actual);

        actual = TimeConverter.durationToString(hoursAndMinutes);
        assertEquals(hoursAndMinutesString, actual);

        actual = TimeConverter.durationToString(zeroDuration);
        assertEquals(zeroDurationString, actual);
    }
}

File: app-v3/src/test/java/com/khmelenko/lab/varis/TestTimeConverter.kt
package com.khmelenko.lab.varis

import com.khmelenko.lab.varis.converter.TimeConverter

import org.junit.Test

import org.junit.Assert.assertEquals

/**
 * Testing TimeConverter class
 *
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
class TestTimeConverter {

    @Test
    fun testDurationToString() {

        val zeroDuration: Long = 0
        val onlySeconds: Long = 45
        val onlyMinutes = (5 * 60).toLong()
        val onlyHours = (3 * 60 * 60).toLong()
        val secondsAndMinutes = onlyMinutes + onlySeconds
        val hoursMinutesAndSeconds = onlyHours + secondsAndMinutes
        val hoursAndSeconds = onlyHours + onlySeconds
        val hoursAndMinutes = onlyHours + onlyMinutes

        val zeroDurationString = "0 sec"
        val onlySecondsString = "45 sec"
        val onlyMinutesString = "5 min"
        val onlyHoursString = "3 hours"
        val secondsAndMinutesString = String.format("%s %s", onlyMinutesString, onlySecondsString)
        val hoursMinutesAndSecondsString = String.format("%s %s %s", onlyHoursString, onlyMinutesString, onlySecondsString)
        val hoursAndSecondsString = String.format("%s %s", onlyHoursString, onlySecondsString)
        val hoursAndMinutesString = String.format("%s %s", onlyHoursString, onlyMinutesString)

        var actual = TimeConverter.durationToString(onlySeconds)
        assertEquals(onlySecondsString, actual)

        actual = TimeConverter.durationToString(onlyMinutes)
        assertEquals(onlyMinutesString, actual)

        actual = TimeConverter.durationToString(onlyHours)
        assertEquals(onlyHoursString, actual)

        actual = TimeConverter.durationToString(secondsAndMinutes)
        assertEquals(secondsAndMinutesString, actual)

        actual = TimeConverter.durationToString(hoursMinutesAndSeconds)
        assertEquals(hoursMinutesAndSecondsString, actual)

        actual = TimeConverter.durationToString(hoursAndSeconds)
        assertEquals(hoursAndSecondsString, actual)

        actual = TimeConverter.durationToString(hoursAndMinutes)
        assertEquals(hoursAndMinutesString, actual)

        actual = TimeConverter.durationToString(zeroDuration)
        assertEquals(zeroDurationString, actual)
    }
}
